mdoc fails to document .NET assemblies

Assemblies: [VSTS](https://apidrop.visualstudio.com/_git/binaries?path=%2Fdotnet&version=GBnpe-2.1)

Latest build: [VSTS](https://apidrop.visualstudio.com/Content%20CI/_build/results?buildId=29399&view=logs)

mdoc release: 5.7.3

Error:

```bash
2018-08-31T22:50:56.6664185Z Updating: System.Net.CookieException
2018-08-31T22:50:56.6721348Z Updating: System.Net.CredentialCache
2018-08-31T22:50:56.6791656Z Updating: System.Net.DecompressionMethods
2018-08-31T22:50:56.6837627Z Updating: System.Net.DnsEndPoint
2018-08-31T22:50:56.6904498Z Updating: System.Net.EndPoint
2018-08-31T22:50:56.6985721Z Updating: System.Net.HttpStatusCode
2018-08-31T22:50:57.7898497Z ##[error]mdoc: System.InvalidCastException: Unable to cast object of type 'System.Xml.XmlComment' to type 'System.Xml.XmlElement'.
   at System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at Mono.Documentation.MDocUpdater.WriteElement(XmlNode parent, String element, Boolean forceNewElement)
   at Mono.Documentation.MDocUpdater.UpdateType(XmlElement root, TypeDefinition type, FrameworkTypeEntry typeEntry)
   at Mono.Documentation.MDocUpdater.DoUpdateType2(String message, XmlDocument basefile, TypeDefinition type, FrameworkTypeEntry typeEntry, String output, Boolean insertSince)
   at Mono.Documentation.MDocUpdater.DoUpdateType(AssemblySet set, TypeDefinition type, FrameworkTypeEntry typeEntry, String basepath, String dest)
   at Mono.Documentation.MDocUpdater.DoUpdateAssembly(AssemblySet assemblySet, AssemblyDefinition assembly, XmlElement index_types, String source, String dest, HashSet`1 goodfiles)
   at Mono.Documentation.MDocUpdater.DoUpdateAssemblies(String source, String dest)
   at Mono.Documentation.MDocUpdater.Run(IEnumerable`1 args)
   at Mono.Documentation.MDoc.Main(String[] args)
See `mdoc help' for more information or use --debug to diagnose.
```

--- a/mdoc/Mono.Documentation/MDocUpdater.cs
+++ b/mdoc/Mono.Documentation/MDocUpdater.cs
@@ -1334,7 +1334,7 @@ namespace Mono.Documentation
                                 var styledNodes = doc.SelectNodes ("//*[@apistyle='" + style.ToString ().ToLowerInvariant () + "']");
                                 if (styledNodes != null && styledNodes.Count > 0)
                                 {
-                                    foreach (var node in styledNodes.Cast<XmlNode> ())
+                                    foreach (var node in styledNodes.SafeCast<XmlNode> ())
                                     {
                                         node.ParentNode.RemoveChild (node);
                                     }
@@ -2824,7 +2824,7 @@ namespace Mono.Documentation
 
         internal static XmlElement WriteElement (XmlNode parent, string element, bool forceNewElement = false)
         {
-            XmlElement ret = parent.ChildNodes.Cast<XmlElement>().FirstOrDefault(e => e.LocalName == element); //(XmlElement)parent.SelectSingleNode (element);
+            XmlElement ret = parent.ChildNodes.SafeCast<XmlElement>().FirstOrDefault(e => e.LocalName == element);
             if (ret == null || forceNewElement)
             {
                 string[] path = element.Split ('/');
@@ -3372,7 +3372,7 @@ namespace Mono.Documentation
                 .Distinct () // make sure there aren't any dupes
                 .ToDictionary (a => a, a => false); // key is the attribute, value is whether it was found
 
-            var attributesState = e.ChildNodes.Cast<XmlElement> ()
+            var attributesState = e.ChildNodes.SafeCast<XmlElement> ()
                                 .Select (elem => new
                                 {
                                     ExistingValue = elem.FirstChild.InnerText,
@@ -3431,7 +3431,7 @@ namespace Mono.Documentation
 
             // clean up
             if (fx.IsLastFramework) {
-                foreach(var attr in e.ChildNodes.Cast<XmlElement> ().ToArray()) {
+                foreach(var attr in e.ChildNodes.SafeCast<XmlElement> ().ToArray()) {
                     if (attr.HasAttribute (Consts.FrameworkAlternate))
                     {
                         var fxAttributeValue = attr.GetAttribute (Consts.FrameworkAlternate);

--- a/mdoc/Mono.Documentation/Updater/DocUtils.cs
+++ b/mdoc/Mono.Documentation/Updater/DocUtils.cs
@@ -102,6 +102,19 @@ namespace Mono.Documentation.Updater
             }
         }
 
+        public static IEnumerable<T> SafeCast<T> (this System.Collections.IEnumerable list)
+        {
+            if (list == null) yield break;
+
+            foreach (object item in list)
+            {
+                if (item is T castedItem)
+                {
+                    yield return castedItem;
+                }
+            }
+        }
+
         public static bool IsExplicitlyImplemented (MethodDefinition method)
         {
             return method != null && method.IsPrivate && method.IsFinal && method.IsVirtual;

